source/api.py

```diff
@@ -317,8 +317,15 @@ def getStatusBar():
 	if not location:
 		return None
 	left, top, width, height = location
-	bottom = top + height - 1
-	obj = getDesktopObject().objectFromPoint(left, bottom)
+	x=1
+	while x<12:
+		bottom = top + height - x
+		if left<0:
+			left=0

```

Comment:
Yes this was done because there were cases for which left was coming out to
be a negative number.

For maximized Visual Studio window, the co-ordinates come out to be:
Left: -8
Top: -8
Height: 1096

In maximized view, the existing solution was not giving status bar but when
I changed bottom coordinate was changed to bottom=top+height-11, we could
get the status bar correctly.

For non maximized Visual Studio window, the left, top and height
coordinates vary depending on the window size. Also the solution was
working as it is without any change, i.e.,
bottom=top+height-1

Regards
Manshul V Belani

On Wed, Jan 13, 2016 at 7:19 AM, James Teh <notifications@github.com> wrote:

> In source/api.py
> <https://github.com/nvaccess/nvda/pull/5620#discussion_r49542751>:
>
> > @@ -317,8 +317,15 @@ def getStatusBar():
> >  	if not location:
> >  		return None
> >  	left, top, width, height = location
> > -	bottom = top + height - 1
> > -	obj = getDesktopObject().objectFromPoint(left, bottom)
> > +	x=1
> > +	while x<12:
> > +		bottom = top + height - x
> > +		if left<0:
> > +			left=0
>
> I assume this is because you were seeing cases where left was a negative
> number? When?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nvaccess/nvda/pull/5620/files#r49542751>.
>



-- 
Regards
Manshul V Belani